crates/core/liveview-native-core-swift/Sources/LiveViewNativeCore/Support.swift

```diff
@@ -0,0 +1,147 @@
+import Foundation
+
+class SimpleHandler: DocumentChangeHandler {
+    var callback: (Document, NodeRef) -> ()
+    init (
+        _ callback: @escaping (Document, NodeRef) -> ()
+    ) {
+       self.callback = callback
+    }
+    func handle(_ context: Document, _ changeType: ChangeType, _ node: NodeRef, _ parent: NodeRef?) {
+        switch changeType {
+        case .add:
+            self.callback(context, parent!)
+        case .remove:
+            self.callback(context, parent!)
+        case .change:
+            self.callback(context, node)
+        case .replace:
+            self.callback(context, parent!)
+        }
+    }
+
+}
+
+extension Document {
+    public subscript(ref: NodeRef) -> Node {
+        let data = self.get(ref)
+        return Node(self, ref, data)
+    }
+    public static func parseFragmentJson(payload: [String: Any]) throws -> Document {
+        let jsonData = try JSONSerialization.data(withJSONObject: payload, options: .prettyPrinted)
+        let payload = String(data: jsonData, encoding: .utf8)!
+        return try Document.parseFragmentJson(payload)
+    }
+    public func mergeFragmentJson(
+        _ payload: [String: Any]
+        ) throws {
+        let jsonData = try JSONSerialization.data(withJSONObject: payload, options: .prettyPrinted)
+        let payload = String(data: jsonData, encoding: .utf8)!
+
+        return try self.mergeFragmentJson(payload)
+    }
+
+    public func on(_ event: EventType, _ callback: @escaping (Document, NodeRef) -> ()) {
+
+        let simple = SimpleHandler(callback)
+        self.setEventHandler(simple)
+    }
+}
+
+extension AttributeName: ExpressibleByStringLiteral {
+    public init(stringLiteral value: String) {
+        self.init(namespace: .none, name: value)
+    }
+    public var rawValue: String {
+        if let namespace {
+            return "\(namespace):\(name)"
+        } else {
+            return name
+        }
+    }
+    public init?(rawValue: String) {
+        let parts = rawValue.split(separator: ":")
```

Comment:
If you enforce the separator to be a single `Character` this is a bit more performant.
```suggestion
        let parts = rawValue.split(separator: ":" as Character)
```